src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -789,75 +808,73 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
-    LOG4CXX_WARN(logger_, "Application types are not an array!");
-    return;
-  }
-
-  const SmartObject& app_type_so = app_type->getElement(0);
-  if (NsSmartDeviceLink::NsSmartObjects::invalid_object_value == app_type_so) {
-    LOG4CXX_WARN(logger_, "Invalid object for component type!");
+  if (not_responding_interfaces.empty()) {
+    LOG4CXX_DEBUG(logger_,
+                  "not_responding_interfaces is empty, nothing to do. return.");
     return;
   }
 
-  const int64_t app_hmi_type = app_type_so.asInt();
-  const std::string& component_name = AppHMITypeToString(
-      static_cast<mobile_apis::AppHMIType::eType>(app_hmi_type));
-  const std::string component_info =
-      component_name + " component does not respond";
-  (*response)[strings::msg_params][strings::info] = component_info;
-}
-
-std::string CommandRequestImpl::AppHMITypeToString(
-    const mobile_apis::AppHMIType::eType app_hmi_type) const {
-  switch (app_hmi_type) {
-    case mobile_apis::AppHMIType::DEFAULT:
-      return "Default";
-    case mobile_apis::AppHMIType::COMMUNICATION:
-      return "Communication";
-    case mobile_apis::AppHMIType::MEDIA:
-      return "Media";
-    case mobile_apis::AppHMIType::MESSAGING:
-      return "Messaging";
-    case mobile_apis::AppHMIType::NAVIGATION:
-      return "Navigation";
-    case mobile_apis::AppHMIType::INFORMATION:
-      return "Information";
-    case mobile_apis::AppHMIType::SOCIAL:
-      return "Social";
-    case mobile_apis::AppHMIType::BACKGROUND_PROCESS:
-      return "Background Process";
-    case mobile_apis::AppHMIType::TESTING:
-      return "Testing";
-    case mobile_apis::AppHMIType::SYSTEM:
-      return "System";
-    default:
-      return "Invalid Component Type";
+  // We copy contents to vector so we can have iterator that will allow us to do
+  // operator-() on it, it will be used in the check later
+  std::vector<HmiInterfaces::InterfaceID> not_responding_interfaces_vector;
+
+  std::copy(not_responding_interfaces.begin(),
+            not_responding_interfaces.end(),
+            std::back_inserter(not_responding_interfaces_vector));
+  LOG4CXX_DEBUG(logger_,
+                "interfaces_of_interest.size()"
+                    << not_responding_interfaces.size());
+  std::string not_responding_interfaces_string;
+  for (std::vector<HmiInterfaces::InterfaceID>::const_iterator it =
+           not_responding_interfaces_vector.begin();
+       it != not_responding_interfaces_vector.end();
+       ++it) {
+    const HmiInterfaces::InterfaceID& not_responding_interface = *it;
+    const std::string component_name =
+        GetComponentNameFromInterface(not_responding_interface);
+    not_responding_interfaces_string += component_name;
+    if (it != (std::end(not_responding_interfaces_vector) - 1)) {
+      not_responding_interfaces_string += ", ";
+    }
+  }
+  LOG4CXX_DEBUG(
+      logger_,
+      "not_responding_interfaces_string: " << not_responding_interfaces_string);
```

Comment:
@pold500 Change message for more readable: e.g. `"Not respond : "<< not_responding_interfaces_string` or print the same string as component_info